config.js

```diff
@@ -1,106 +1,132 @@
 /* jshint asi: true, node: true, laxbreak: true, laxcomma: true, undef: true, unused: true, esversion: 6 */
 
-const perServiceEnvs = ['MONGODB_URI', 'GITHUB_CLIENT_ID', 'GITHUB_CLIENT_SECRET', 'SLACK_CHANNEL', 'SLACK_ICON_URL']
-if (process.env.SERVICE === 'balance') {
-  process.env.PORT = process.env.PORT || 3003
-  perServiceEnvs.forEach(function(baseEnv) {
-    process.env[baseEnv] = process.env[baseEnv] || process.env['BALANCE_' + baseEnv]
-  })
-} else if (process.env.SERVICE === 'eyeshade') {
-  process.env.PORT = process.env.PORT || 3002
-  perServiceEnvs.forEach(function(baseEnv) {
-    process.env[baseEnv] = process.env[baseEnv] || process.env['EYESHADE_' + baseEnv]
-  })
-} else if (process.env.SERVICE === 'ledger') {
-  process.env.PORT = process.env.PORT || 3001
-  perServiceEnvs.forEach(function(baseEnv) {
-    process.env[baseEnv] = process.env[baseEnv] || process.env['LEDGER_' + baseEnv]
-  })
+const services = {
+  ledger: {
+    portno: 3001,
+
+    f: () => {
+      module.exports.wallet.settlementAddress =
+      { BAT : process.env.BAT_SETTLEMENT_ADDRESS                || '0x7c31560552170ce96c4a7b018e93cddc19dc61b6'
+      }
+
+      if (process.env.COINBASE_WIDGET_CODE) {
+        module.exports.wallet.coinbase = { widgetCode : process.env.COINBASE_WIDGET_CODE }
+      }
+
+      helper()
+      uphold()
+    }
+  },
+  
+  eyeshade: {
+    portno: 3002,
+
+    f: () => {
+      if (process.env.PUBLISHERS_URL) {
+        module.exports.publishers =
+          { url                 : process.env.PUBLISHERS_URL    || 'http://127.0.0.1:3000'
+          , access_token        : process.env.PUBLISHERS_TOKEN  || '00000000-0000-4000-0000-000000000000'
+          }
+      }
+
+      helper()
+      uphold()
+    }
+  },
+
+  balance: {
+    portno: 3003,
+
+    f: () => {
+      if (process.env.LEDGER_URL) {
+        module.exports.ledger = { url : process.env.LEDGER_URL  || 'http://127.0.0.1:3001' }
+      }
+
+      helper()
```

Comment:
we still need uphold config if we are supporting the card based apis for older builds

Reply:
good point.